crates/bevy_ecs/src/system/system_param.rs

```diff
@@ -210,6 +210,75 @@ fn assert_component_access_compatibility(
            query_type, filter_type, system_name, accesses);
 }
 
+/// A collecton of potentially conflicting [`SystemParam`]s allowed by disjoint access.
+///
+/// Allows to safely access and interact with up to 8 mutually exclusive [`SystemParam`]s such as
+/// two queries that reference the same mutable data  or a mutable query and acess to the ecs world.
```

Comment:
```suggestion
/// two queries that reference the same mutable data or an event reader and writer of the same type.
```
This is a much more common use case, so it should be surfaced here.